extensions/che-theia-hosted-plugin-manager-extension/src/node/che-workspace-hosted-plugin-uri-post-processor.ts

```diff
@@ -0,0 +1,79 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { injectable } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+import { HostedPluginUriPostProcessor } from '@theia/plugin-ext';
+import WorkspaceClient, { IRemoteAPI, IRestAPIConfig } from '@eclipse-che/workspace-client';
+import { che } from '@eclipse-che/api';
+
+@injectable()
+export class CheWorkspaceHostedPluginUriPostProcessor implements HostedPluginUriPostProcessor {
+    protected restApiClient: IRemoteAPI;
+
+    constructor() {
+        const restAPIConfig: IRestAPIConfig = {};
+        restAPIConfig.baseUrl = process.env.CHE_API;
+        const token = process.env.CHE_MACHINE_TOKEN;
+        if (token) {
+            restAPIConfig.headers = {};
+            restAPIConfig.headers['Authorization'] = token;
+        }
+        this.restApiClient = WorkspaceClient.getRestApi(restAPIConfig);
```

Comment:
`@eclipse-che/plugin` is not providing now the access to workspace ?

Reply:
`@eclipse-che/plugin` provide API for plugins, `che-theia-hosted-plugin-manager-extension` is extension, so no it not. But I can reuse `@eclipse-che/theia-plugin-ext` to get current workspace